### ISSUE TITLE
File won't open

### ISSUE THREAD

**Issue by ChMah** [jgraph/drawio-desktop#1262]
### Preflight Checklist

* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio-desktop/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

You must agree to search and the code of conduct. You must fill in this entire template. If you delete part/all or miss parts out your issue will be closed.

**Describe the bug**
One of my files doesn't open with the diagram.net app, although it usually does.

**To Reproduce**

I am truly sorry but I don't get your point. What is " '...' " ?

**Expected behavior**
I would like to open my file with the diagram.net app on my desk.

**Screenshots**
Here is the screenshot of the error message when I try to open: 
<img width="408" alt="Capture d’écran 2023-03-17 à 09 37 14" src="https://user-images.githubusercontent.com/128132827/225854639-9df3d5e2-63ef-411d-9b31-c7830a44cceb.png">

**draw.io version (In the Help->About menu of the draw.io editor):**

- draw.io version 20.8.16

**Desktop (please complete the following information):**
 - OS: MacOS Ventura 13.2.1

**Additional context**
I loaded it on Sharepoint and tried to open it with the app, and with the online version (diagrams.net), it doesn't work either. The error message is the same.
I am ok with giving you the file if needed, but I would prefer to send the file rather than giving a link to OneDrive (it's the OneDrive of my organization). How can I do that ?

**Comment by mararad** [comment]
Hi,

When you look at the file version history, are you able to find an older version which can be opened correctly? 
If not, please use support email from https://github.com/jgraph/drawio/wiki/Getting-Support

**Comment by ChMah** [comment]
Hi mararad, thank you for your answer.

I don't know how to look at the file history, but I did save a version before making big changes (opening several tabs and dispersing what was first on a single page). Would it fit ?

**Comment by davidjgraph** [comment]
Do you have file history where you saved the file? If yes, please restore the previous version of the page.

**Comment by ChMah** [comment]
Hi Davidjgraph, I don't have file history when I save files. I work with the app directly on my desk, not on a web browser. I did activate the automatic savings, but I can't find any saving folder with previous version of my file - so I guess each saving replace the previous one.
I found though a saved version of my file more recent than the one I told you about in my previous message. Here is the [link](https://drive.google.com/file/d/1ui805tFQk3C2mS5n-6P8gHZTGxEXlCai/view?usp=sharing).

**Comment by Narvath** [comment]
@ChMah With the latest Draw.io version (21.4.0) I could succesfully open the diagram you provided in your last post.

**Comment by ChMah** [comment]
Hi Narvath, thank your for trying. The diagram provided in the last post opens for me too. Someone asked me a working version. You can find [here](https://drive.google.com/file/d/1-skeWYny4NxVfgHFcOMe-xJqvjYUzD4T/view?usp=sharing) the file that doesn't open.
Is it working for you?

**Comment by Narvath** [comment]
@ChMah J'ai réussi à l'ouvrir en rajoutant les tags  
`  </diagram>
</mxfile>`
à la fin du fichier.
La structure du fichier ne suit pas celle normal, il manque une tonne de tags/divisions et j'imagine que c'est pour cela qu'il est impossible de l'enregistrer une fois ouvert. Il devrait toutefois être possible de copier/coler les éléments qu'il contient dans un nouveau fichier puis continuer à partir de la.

[Mutualisation des ressources copie_2023,03,17.txt](https://github.com/jgraph/drawio-desktop/files/11789264/Mutualisation.des.ressources.copie_2023.03.17.txt)

Change l'extension .txt du fichier que je t'ai envoyé par un .drawio (Github n'accepte pas les .drawio)

En esérant que ça aide!

I managed to open it by adding the tags
`  </diagram>
</mxfile>`
at the end of the file.
The file structure doesn't follow the standard one; there are missing a lot of tags/divisions, and I assume that's why it's impossible to save once opened. However, it should be possible to copy/paste the elements it contains into a new file and continue from there.

Change the file extension from .txt to .drawio for the file I sent you (GitHub doesn't accept .drawio files).

Hope it helps!

**Comment by ChMah** [comment]
Hi Narvath, thank your for trying. The diagram provided in the last post opens for me too. Someone asked me a working version. You can find [here](https://drive.google.com/file/d/1-skeWYny4NxVfgHFcOMe-xJqvjYUzD4T/view?usp=sharing) the file that doesn't open.
Does it work for you?

**Comment by ChMah** [comment]
Magnifique ! Ca fonctionne !
J'ai d'autres fichiers qui ne fonctionnent plus. Peux-tu m'indiquer comment (et où surtout) tu as ajouté les tags </diagram> </mxfile> ?

**Comment by Narvath** [comment]
@ChMah 
Bien sur,
Premièrement, le message d'erreur lors de l'ouverture pointe où est le problème. De mes observations, c'est toujours à la fin du code. J'ai regardé le code d'un projet qui ouvre pour voir comment était la structure puis j'ai rajouté les tags manquants à la fin du fichier corrompu. Dans un de mes fichier, j'ai du supprimer le dernier élément qui n'était pas complet (une mxCell) (Il manquait des paramètres comparativement aux autres et n'avait pas de '>' à la fin) en plus de rajouter les tags manquants. Dans ton fichier, le dernier élément était complet.
J'ai aussi utilisé ce petit site web https://www.aliciaramirez.com/closing-tags-checker/ pour vérifier si je n'avait pas oublié de tag.

Sure,
Firstly, the error message upon opening points to where the problem lies. From my observations, it's always at the end of the code. I looked at the code of a project that opens successfully to see how the structure was, and then I added the missing tags at the end of the corrupted file. In one of my files, I had to remove the last element that wasn't complete (an mxCell) (it was missing some parameters compared to the others and didn't have a '>' at the end) in addition to adding the missing tags. 
In your file, the last element was complete.
I also used this little website, https://www.aliciaramirez.com/closing-tags-checker/, to check if I had missed any tags.
